.golangci.yml

```diff
@@ -1,164 +1,150 @@
-        - ireturn          # we have exceptions, so need to silence them in tests
-        - lll              # tests often require long lines
-        - maintidx         # not critical for tests
-        - nonamedreturns   # not critical for tests, albeit desirable
-        - testpackage      # senseless
-        - unused           # very annoying false positive: https://github.com/golangci/golangci-lint/issues/791
-    - path: managed/models/
-      linters:
-        - recvcheck
+    - varnamelen
+    - wastedassign
+    - wrapcheck
+    - wsl
+  settings:
+    cyclop:
+      max-complexity: 30
+    depguard:
+      rules:
+        main:
+          deny:
+            - pkg: github.com/pkg/errors
+              desc: use "errors" instead
+            - pkg: github.com/gogo/protobuf/proto
+              desc: use "github.com/golang/protobuf/proto" instead
+            - pkg: github.com/percona/go-mysql/log/slow
+              desc: only use the forked parser
+            - pkg: github.com/percona-platform/platform
+              desc: use "github.com/percona/saas" instead
+            - pkg: github.com/percona/platform
+              desc: use "github.com/percona/saas" instead
+            - pkg: gopkg.in/yaml.v2
+              desc: use "gopkg.in/yaml.v3" instead
+    exhaustive:
+      default-signifies-exhaustive: true
+    godot:
+      scope: toplevel
+      exclude:
+        - go-sumtype:decl
+        - pmm-managed
+      capital: true
+      period: true
+    inamedparam:
+      skip-single-param: true
+    ireturn:
+      allow:
+        - anon
+        - error
+        - empty
+        - stdlib
+        - github.com/charmbracelet/bubbletea.Model
+        - github.com/percona/pmm/admin/commands.Result
+        - github.com/percona/pmm/agent/runner/actions.Action
+        - github.com/percona/pmm/managed/services/telemetry.DataSource
+    lll:
+      line-length: 170
+      tab-width: 4
+    maintidx:
+      under: 20
+    nestif:
+      min-complexity: 7
+    recvcheck:
+      exclusions:
+        - '*.pb.go'
+        - '*_test.go'
+        - mock_*.go
+        - '*_reform.go'
+    tagliatelle:
+      case:
+        rules:
+          avro: snake
+          bson: camel
+          json: ""
+          xml: camel
+          yaml: ""
+    unparam:
+      check-exported: true
+  exclusions:
+    generated: lax
+    rules:
+      - linters:
+          - contextcheck
+          - exhaustivestruct
+          - forcetypeassert
+          - funlen
+          - gocognit
+          - goconst
+          - ireturn
+          - lll
+          - maintidx
+          - mnd
+          - nonamedreturns
+          - testpackage
+          - unused
+        path: _test\.go
+      - linters:
+          - recvcheck
+        path: managed/models/
+      - path: (.+)\.go$
+        text: 'G104: Errors unhandled'
+      - path: (.+)\.go$
+        text: exported method `.+` should have comment or be unexported
```

Comment:
I'm surprised why do we exclude it

Reply:
I'm confussed by this [rule](https://github.com/percona/pmm/pull/4289/files#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9R107-R109) which contains path out of those list. If it works then other ones applies for all files in repo.

Reply:
I removed all those exclusions, will keep an eye on v3 and propose changes as needed.